src/bundles/pinning.js

```diff
@@ -55,23 +62,83 @@ const uniqueCidBatches = (arrayOfCids, size) => {
   return result
 }
 
+const remotePinLs = (ipfs, params) => {
+  const backoffs = readSetting('remotesServicesBackoffs') || {}
+  const { service } = params
+
+  const { lastTry, tryAfter } = backoffs[service] || { lastTry: 0, tryAfter: PIN_CHECK_INTERVAL }
+  if (lastTry + tryAfter > new Date().getTime()) {
+    throw new Error('still within back-off period')
+  }
+
+  try {
+    return ipfs.pin.remote.ls(params)
+  } catch (e) {
+    if (e.toString().includes('429 Too Many Requests')) {
```

Comment:
Some logic in the pinning service compliance checker shows how to read the headers and automatically await the appropriate time (retry header, tokens available, etc.) I believe its in the middleware.

Instead of reaching the point of failure and catching it, we should be preventing the exhausting of the available calls from the get-go.

Reply:
Tracked in #2043